pkg/apis/componentconfig/types.go

```diff
@@ -160,58 +161,115 @@ const (
 	ProxyModeIPTables  ProxyMode = "iptables"
 )
 
+// SchedulerPolicyConfigMapKey defines the key of the element in the
+// scheduler's policy ConfigMap that contains scheduler's policy config.
+const SchedulerPolicyConfigMapKey string = "policy.cfg"
+
+// SchedulerPolicySourceType define ways to obtain a scheduler Policy at
+// runtime.
+type SchedulerPolicySourceType string
+
+const (
+	// SchedulePolicySourceTypeFile means get a Policy from
+	// a SchedulerPolicyFileSource.
+	SchedulePolicySourceTypeFile SchedulerPolicySourceType = "file"
+	// SchedulerPolicySourceTypeConfigMap means get a Policy from a
+	// SchedulerPolicyConfigMapSource.
+	SchedulerPolicySourceTypeConfigMap SchedulerPolicySourceType = "configmap"
+)
+
+// SchedulerPolicySource configures a means to obtain a scheduler Policy. Given
+// a SchedulerPolicySourceType, the related source should be used. Sources are
+// mutually exclusive.
+type SchedulerPolicySource struct {
+	// Type is the policy source type.
+	Type SchedulerPolicySourceType
+	// File is a file policy source.
+	File *SchedulerPolicyFileSource
+	// ConfigMap is a config map policy source.
+	ConfigMap *SchedulerPolicyConfigMapSource
+}
+
+// SchedulerPolicyFileSource is a policy serialized to disk and accessed via
+// path.
+type SchedulerPolicyFileSource struct {
+	// Path is the location of a serialized policy.
+	Path string
+}
+
+// SchedulerPolicyConfigMapSource is a policy serialized into a config map value.
+type SchedulerPolicyConfigMapSource struct {
+	// Reference is a config map containing a seralized policy in the
+	// SchedulerPolicyConfigMapKey key.
+	Reference api.ObjectReference
```

Comment:
I think we've agreed not to depend on this since the type is overly large and actually wrong about depending on things like kind vs. resources.  Specific types for specific cases as I recall brian's advice, right @liggitt ?

Reply:
yeah, use-specific object

Reply:
I removed the use of ObjectReference.